lib/libc/common/Kconfig

```diff
@@ -65,3 +65,9 @@ config COMMON_LIBC_STRNLEN
 	bool
 	help
 	  common implementation of strnlen().
+
+config COMMON_LIBC_CLOCK
+	bool "Common C library clock"
+	default y
```

Comment:
This should not be `default y` (none of the configs that do not depend on other configs here should be); instead, it should be selected by each libc that needs this.

Reply:
Updated for minimal, newlib, and picolibc
cc @evgeniy-paltsev